firefox-ios/Client/Assets/CC_Script/TranslationsHelpers.mjs

```diff
@@ -0,0 +1,135 @@
+/// attribute that returns the parent node in the flattened DOM tree.
+///
+/// In normal web JS environments, the flattened tree parent is equivalent
+/// to the `parentElement` property, so we can safely emulate this by
+/// returning `this.parentElement`.
+Object.defineProperty(Node.prototype, "flattenedTreeParentNode", {
+  get() {
+    return this.parentElement ?? null;
+  },
+  configurable: true,
+});
+
+export const Cu = {
+  isDeadWrapper: (node) => !node?.isConnected,
+  isInAutomation: false,
+};
+globalThis.Cu = Cu;
+
+/// Polyfill for Gecko's non-standard DOMParser.parseFromSafeString.
+/// In Gecko, this ensures the parsed document uses the same principal
+/// as the owner global.
+/// For normal web JS, we simply forward to parseFromString.
+DOMParser.prototype.parseFromSafeString = function (str, type) {
+  return this.parseFromString(str, type);
+};
+
+export const setTimeout = globalThis.setTimeout.bind(window);
+export const clearTimeout = globalThis.clearTimeout.bind(window);
+
+/// NOTE: Polyfill for Gecko's non-standard console.createInstance.
+/// For reference, see:
+/// - https://firefox-source-docs.mozilla.org/toolkit/javascript-logging.html
+globalThis.console.createInstance = () => ({
+  log: (...params) => console.log("createInstance --- ", ...params),
+  warn: (...params) => console.warn("createInstance --- ", ...params),
+  error: (...params) => console.error("createInstance --- ", ...params),
+  shouldLog: () => false,
+});
+
+globalThis.TE_getLogLevel = () => { };
+globalThis.TE_log = (message) => console.log("TE_log ---- ", message);
+globalThis.log = (message) => console.log("log ---- ", message);
+globalThis.TE_logError = (...error) =>
+  console.error("TE_error ---- ", ...error);
+
+/// NOTE: Polyfills for various ChromeUtils and TE_* functions used by
+/// translations-engine.js in Gecko.
+globalThis.ChromeUtils = globalThis.ChromeUtils || {};
+globalThis.ChromeUtils.addProfilerMarker = () => { };
+globalThis.ChromeUtils.domProcessChild = {
+  getActor: () => globalThis,
+};
+globalThis.ChromeUtils.now = () => performance.now();
+globalThis.Services = { intl: { getScriptDirection: () => "ltr" } };
+
+
+/// NOTE: Stubs for TE_* functions used by translations-engine.js in Gecko.
+/// These are no-ops or empty implementations as needed (for now).
+globalThis.TE_getLogLevel = () => { };
+globalThis.TE_destroyEngineProcess = () => { };
+globalThis.TE_reportEnginePerformance = () => { };
+globalThis.TE_reportEngineStatus = () => { };
+globalThis.TE_resolveForceShutdown = () => { };
+globalThis.TE_addProfilerMarker = () => { };
+/// NOTE: Implementation for TE_requestEnginePayload used by translations-engine.js in Gecko.
+/// This fetches the translation models and translator WASM from the app's
+/// translations:// URLs and returns them in the expected format.
+globalThis.TE_requestEnginePayload = async ({ sourceLanguage, targetLanguage }) => {
```

Comment:
@nbhasin2 this is the important method that loads the lang models and the translator. The other polyfills are mostly no-ops.